### DISCUSSION TITLE
Looking for a good `[project.entry-point]`

### DISCUSSION THREAD

**Discussion by joshuadavidthomas** [ehmatthes/django-simple-deploy#314] [General]
I really appreciate how you've approached working in public @ehmatthes and laid all the bread crumbs out about where the project needs to go in the GitHub issues. The problem is given the volume of issues, I'm having a bit of trouble finding a good place to start with helping out and what is still relevant.

I know the project is in a bit of a transition period, but I was curious if you had any specific issues you've been itching to tackle that you could use some help on right now wouldn't step on any future plans?

(Couldn't resist the Python packaging pun with the title. 😁)

**Comment by ehmatthes** [comment]
> The problem is given the volume of issues, I'm having a bit of trouble finding a good place to start with helping out and what is still relevant.

Yes, I completely agree. Next time I start a project I'll make a notes file for anything that isn't a closable task. I should be able to continue working on this project more regularly for a while, and I'm going to keep closing issues that are too vague to be actionable.

There are a couple places I think we can start to collaborate more meaningfully:

- If you want to focus on any of the platforms that are already supported, any work you do to improve the current process would be helpful, and would align well with ongoing work. There's pretty good isolation between the different modules right now. So if you wanted to help refine the Fly deployment process for example, most of that work should be entirely within the `simple_deploy/management/commands/fly_io/` directory. That code will be moved to a plugin, but that shouldn't really affect the platform-specific deployment code.

- The CLI should be reviewed and updated so it's consistent with default Django management commands. That code is pretty well isolated as well.

Does either of those sound interesting to you?

**Reply by ehmatthes** [reply]
I'll add more entry points in replies, rather than editing the original post. Another specific task someone can take on is #291. I don't have it labeled as 1.0 because it's not critical, but it would be helpful. I think it would cut the local test suite time significantly, which is currently around 20s on my desktop.

It might require cleaning up my inconsistent venv naming conventions over the last three years. I used `dsd_env`, `b_env`, and `.venv`. I definitely want to collapse all those to `.venv`. That's probably a global find and replace, but I'm not sure if it would break anything.

**Reply by joshuadavidthomas** [reply]
> Yes, I completely agree. Next time I start a project I'll make a notes file for anything that isn't a closable task. I should be able to continue working on this project more regularly for a while, and I'm going to keep closing issues that are too vague to be actionable.

I didn't intend for this post to be indictment on your project management as you've been actively working on it and how you have used the repo's issue tracker and I hope it didn't come off that way. You should see some of the issue backlogs of my work repos, except you won't! 😂 Because fortunately for me they are are private and will remain that way. (Ask @jefftriplett, he knows the horrors of which I speak...)

This was more just a call of "I want to help! What is still relevant? Point towards the issues that are top of mind right now that you wish you had more time for." Even old issues or groups of issues that could be reviewed to see if they are still needed or if they can be closed. 

Though I realize even these types of questions still puts a burden on you as the maintainer to go through the issues and think through each one somewhat deeply about: 

- Is this still relevant? 
- Is this a priority right now or can it go on the backburner? 
- Is this even a good issue for someone relatively unfamiliar with how the code base is at it's current state to even tackle?

> Does either of those sound interesting to you?

It all sounds interesting! I love the idea of making deployment of Django projects easier for people not comfortable with it (or even people who are who just want the easy button to press) and want to help out in the best way I can.

I also recognize this is your project and that I am just an outside observer who hasn't necessarily followed all the developments it has gone through. I want to follow your lead as you clearly have put a lot of time and thought in to it.

If it would be helpful, I'm more than happy to review some of the older issues and do some triage to see which ones might still be relevant. I could either comment on each one or write up a list and share it here in this discussion or via email.

**Reply by joshuadavidthomas** [reply]
![meme](https://github.com/ehmatthes/django-simple-deploy/assets/19896267/f486b1c4-9e94-4ab0-9fe7-d13a581b19cc)

**Reply by ehmatthes** [reply]
No hard feelings at all! The volume of issues is ~3 years of thoughts jotted down when I had a week or two to throw at this. But if it's going to go somewhere meaningful, I need to pare all those down and make it a much friendlier project environment for people to walk into and find a way to contribute to. I should have some time to give to this consistently for a while now, and I have a much clearer vision of how to proceed at this point.

I don't think triaging issues will be a good use of your time. I'm going to try to close a bunch over the next few weeks, and I have the context to do that a lot more efficiently than you or anyone else would. I think the best places to contribute currently are:

- Focus on a specific platform's deployment process. This is great if you know a platform well, or want to help maintain that platform long-term.
- Review the CLI and make a plan for how to align it better with standard Django management commands.
- Modify the integration tests to use uv instead of pip, and see how much faster they run.
- Make all the virtual environment names `.venv`, instead of `dsd_env` and `b_env`.

Each of these can be done without having to wade through a bunch of context in stale issues. You're welcome to help in any way you want, just let me know and I'll do the other refactoring in a way where we're not getting in each other's way. If you don't touch this until it's a bit cleaner, that's quite fine as well. This conversation with you is also helping me frame docs/issues for helping others contribute as well.

**Reply by ehmatthes** [reply]
I moved all stale issues to a [Parking Lot](https://django-simple-deploy.readthedocs.io/en/latest/contributing/parking_lot/) section of the docs. All open issues have the 1.0 tag, and they all fit on one page. :)

**Comment by ehmatthes** [comment]
I think the entry points are much clearer now. If not, please feel free to reopen this.
